evm/rlp/headers.py

```diff
@@ -229,10 +224,12 @@ def hash(self):
     @classmethod
     @to_dict
     def _deserialize_header_bytes_to_dict(cls, header_bytes):
-        header_sedes = cls.get_sedes()
         # assume all fields are padded to 32 bytes
         obj_size = 32
-        assert len(header_bytes) == obj_size * len(header_sedes)
+        if len(header_bytes) != obj_size * len(cls.fields):
+            raise ValidationError(
+                "Length of header bytes should be equal to obj_size * len(cls.fields)"
```

Comment:
I would actually be very explicit here as it's frustrating to have to go dig when there is code mis-behaving.

```
raise Validation Error(
    "Expected header bytes to be of length: {0}.  Got length {1} instead.\n- {2}".format(
        obj_size * len(cls.fields), len(header_bytes), encode_hex(header_bytes),
    )
)
```

Good error messages are soooo valuable when things start going [pear shaped](https://en.wikipedia.org/wiki/Pear-shaped).

Reply:
Thank you sooo much! I didn't think of this.
I will keep reminding myself with this suggestion.